src/lib/y2storage/storage_manager.rb

```diff
@@ -460,20 +467,64 @@ def configure_snapper?
 
     # Class methods
     class << self
+      # Initializes storage with a specific access mode (read-only or read-write)
+      #
+      # With the default callbacks, the user is asked whether to retry or abort
+      # when lock cannot be acquired.
+      #
+      # @raise [Error] if the requested mode is incompatible with the already
+      #   created instance (i.e., current instance is ro but rw is requested).
+      #
+      # @param mode [Symbol, nil] :ro, :rw. If nil, a default mode is used, see
+      #   {.default_storage_mode}.
+      # @param callbacks [Callbacks::Initialize, nil]
+      #
+      # @return [Boolean] true if the storage instance was correctly created for
+      #   the given mode; false otherwise.
+      def setup(mode: nil, callbacks: nil)
+        mode ||= default_storage_mode
```

Comment:
Why is this done here if it's also done at the beginning of `.instance`? I find it confusing.